packages/dashboards/addon/routes/dashboards/dashboard/view.js

```diff
@@ -180,7 +176,20 @@ export default Route.extend({
 
     this.controller.set('filters', null);
 
-    this._cancelWidgetDataTasks();
     this.set('_widgetDataCache', null);
+  },
+
+  actions: {
+    /**
+     * @override
+     * @action willTransition
+     * @param {Transition} transition
+     */
+    willTransition(transition) {
+      if (transition.targetName !== this.routeName) {
```

Comment:
Does this get fired when a filter updates the query params for the route? if we update a filter while loading, will this cancel loading widgets?

Reply:
Good point, no test picked that up. I was sure when we update the filters we stay in the same route and the cancelation would be in the `model` hook if filters actually changed, but we actually transition to `dashboards.dashboard.index`. Need to make a small change here and add a test to make sure widgets still load, because right now when adding/removing an empty filter, it cancels loading widgets which we render from cache so nothing loads.